internal/impl/generate/tmpl/devcontainerDockerfile.tmpl

```diff
@@ -25,13 +25,19 @@ ENV PATH="/home/${DEVBOX_USER}/.nix-profile/bin:/home/${DEVBOX_USER}/.devbox/nix
 # Step 5: Installing your devbox project
 WORKDIR /code
 RUN sudo chown $DEVBOX_USER:root /code
-COPY devbox.json devbox.json
+{{if .IsDevcontainer}}
+COPY --chown=$DEVBOX_USER:$DEVBOX_USER devbox.json devbox.json
 {{if len .LocalFlakeDirs}}
 # Step 6: Copying local flakes directories
 {{- end}}
 {{range $i, $element := .LocalFlakeDirs -}}
-COPY {{$element}} {{$element}}
+ADD --chown=$DEVBOX_USER:$DEVBOX_USER {{$element}} {{$element}}
 {{end}}
+{{- else}}
+# Add files/directories you want to exclude in .dockerignore
+ADD --chown=$DEVBOX_USER:$DEVBOX_USER . .
```

Comment:
That could simplify the changes in this PR to:

```
{{ if not .IsDevcontainer }}
COPY .dockerignore .
{{ end }}
```

Reply:
@mikeland73 The motivation behind this PR was so that after user runs `devbox generate dockerfile` they don't have to modify the generated Dockerfile to make it build a docker image for their project. If we just copy `.dockerignore` then whoever uses this feature has to 
1. generate the dockerfile
2. build the image
3. see the project files aren't copied over
4. update dockerfile
5. build again

So I thought copying everything with the exception of known unnecessary items that are included in `.dockerignore` is a better experience.

Reply:
Got it, I understand the PR better now, this was less about the dockerignore and more about copying the entire project directory by default. Seems reasonable to me. A few questions:

* Does dev container not need this? (it would simplify the dockerfile template)
* Why `ADD` instead of `COPY`?